File: mt19937ar/mt19937ar_reference_1.c
#include <stdio.h>
#include "mt19937ar.h"

int main(void)
{
    int i;
    unsigned long init[4]={0x123, 0x234, 0x345, 0x456};
    int length=4;
    mt19937ar_t s;
    int j=0;
    init_by_array(&s, init, length);

    for (i=0; i<4; i++) {
      printf("%11lu\n", genrand_int32(&s));
    }
    for (i=0; i<2000; i++) {
      j = j + genrand_int32(&s);
    }
    for (i=0; i<4; i++) {
      printf("%11lu\n", genrand_int31(&s));
    }

    return 0;
}

File: mt19937ar/mt19937ar_reference_1.f90
program example
  use mt19937ar_type_module
  use mt19937ar_module
  use iso_c_binding
  implicit none

  type(mt19937ar_t) :: state

  integer(c_long) :: init(4) = [291, 564, 837, 1110]
  integer :: i
  integer(c_long) :: j

  call init_by_array(state, init)

  do i=1, 4
     write(*,'(i11)') genrand_int32(state)
  end do

  j = 0
  do i=1, 2000
     j = j + genrand_int32(state)
  end do

  do i=1, 4
     write(*,'(i11)') genrand_int31(state)
  end do

end program example
